test/e2e/gang_scheduling/gang_scheduling_test.go

```diff
@@ -520,6 +521,259 @@ var _ = Describe("", func() {
 		Ω(err).NotTo(gomega.HaveOccurred())
 	})
 
+	// Test to verify Gang Apps FIFO order
+	// Create FIFO queue with quota of 300m and 300M
+	// 1. Deploy appA with 1 pod
+	// 2. Deploy appB with gang of 3 pods
+	// 3. Deploy appC with 1 pod
+	// 4. appA = Starting, appB = Accepted (with 2 reserved phs), appC = Accepted
+	// 5. Delete appA
+	// 6. appA = Completing, appB = Running, appC = Accepted
+	It("Verify_GangApp_FIFO_Order", func() {
+		var fifoQName = "fifoq"
+		By(fmt.Sprintf("Creating namespace: %s", fifoQName))
+		fifoQ, nsErr := kClient.CreateNamespace(fifoQName, nil)
+		Ω(nsErr).NotTo(HaveOccurred())
+		Ω(fifoQ.Status.Phase).To(Equal(v1.NamespaceActive))
+		defer func() { Ω(kClient.DeleteNamespace(fifoQName)).NotTo(HaveOccurred()) }()
+
+		// Create appIDs
+		var apps []string
+		for j := 0; j < 3; j++ {
+			id := fmt.Sprintf("app%d-%s", j, common.RandSeq(5))
+			apps = append(apps, id)
+		}
+
+		// Initial allocation to fill ns quota
+		appAllocs := map[string]map[string]int{
+			apps[0]: {"pods": 1, "?": 0},
+			apps[1]: {"pods": 3, "minMembers": 3},
+			apps[2]: {"pods": 1, "minMembers": 0},
+		}
+		// Expected appState progression
+		appStates := map[string][]string{
+			apps[0]: {"Starting", "Completing"},
+			apps[1]: {"Accepted", "Running"},
+			apps[2]: {"Accepted", "Accepted"},
+		}
```

Comment:
As we have discussed: we need to be 100% sure that these states are stable. Please document why we expect these states and why they are stable